### PR TITLE
Use rds module & CMK in complete example

### PR DESCRIPTION
**Pull request by j0hnmwangi**
# Description

This PR aims to use the RDS module in complete example and also use a CMK

## Features/Fixes/Patches/Changes list:

Please check the [CHANGELOG.md](https://github.com/boldlink/terraform-aws-secretsmanager/blob/feature/use-rds-module/CHANGELOG.md#104---2022-08-04)

### PR Owners Checklist:
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Have you checked there aren't open [Pull Requests](../../../pulls) of upstream or parallel code that will cause conflicts?
* [x] Have you updated the `CHANGELOG.md`?
* [x] Have you updated the `README.md`(s)?
* [x] OWNER: Does your submission pass?
    * [x] Pre-commit (attach logs/print screen to this PR)
    * [x] Checkov/terrascan (attach logs/print screen to this PR)
    * [x] Terraform resources examples build/destroy successfully with Terraform 0.14.11 minimum?
* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### PR Reviewers Checklists?
* [ ] Are all your comments/changes resolved?
* [ ] Are you happy with the OWNER checklists and additional information provided?
* [ ] Does your review tests pass?
    * [ ] Terraform resources examples build/destroy successfully with Terraform 0.14.11 minimum?

<details><summary><b>Checkov scan results:</b></summary>

```bash


       _               _              
   ___| |__   ___  ___| | _______   __
  / __| '_ \ / _ \/ __| |/ / _ \ \ / /
 | (__| | | |  __/ (__|   < (_) \ V / 
  \___|_| |_|\___|\___|_|\_\___/ \_/  
                                      
By bridgecrew.io | version: 2.1.32 
Update available 2.1.32 -> 2.1.83
Run pip3 install -U checkov to update 


terraform scan results:

Passed checks: 34, Failed checks: 0, Skipped checks: 1

Check: CKV_AWS_49: "Ensure no IAM policies documents allow "*" as a statement's actions"
	PASSED for resource: aws_iam_policy_document.assume_role_policy
	File: /examples/complete/data.tf:17-29
	Guide: https://docs.bridgecrew.io/docs/bc_aws_iam_43
Check: CKV_AWS_108: "Ensure IAM policies does not allow data exfiltration"
	PASSED for resource: aws_iam_policy_document.assume_role_policy
	File: /examples/complete/data.tf:17-29
	Guide: https://docs.bridgecrew.io/docs/ensure-iam-policies-do-not-allow-data-exfiltration
Check: CKV_AWS_1: "Ensure IAM policies that allow full "*-*" administrative privileges are not created"
	PASSED for resource: aws_iam_policy_document.assume_role_policy
	File: /examples/complete/data.tf:17-29
	Guide: https://docs.bridgecrew.io/docs/iam_23
Check: CKV_AWS_110: "Ensure IAM policies does not allow privilege escalation"
	PASSED for resource: aws_iam_policy_document.assume_role_policy
	File: /examples/complete/data.tf:17-29
	Guide: https://docs.bridgecrew.io/docs/ensure-iam-policies-does-not-allow-privilege-escalation
Check: CKV_AWS_109: "Ensure IAM policies does not allow permissions management / resource exposure without constraints"
	PASSED for resource: aws_iam_policy_document.assume_role_policy
	File: /examples/complete/data.tf:17-29
	Guide: https://docs.bridgecrew.io/docs/ensure-iam-policies-do-not-allow-permissions-management-resource-exposure-without-constraint
Check: CKV_AWS_111: "Ensure IAM policies does not allow write access without constraints"
	PASSED for resource: aws_iam_policy_document.assume_role_policy
	File: /examples/complete/data.tf:17-29
	Guide: https://docs.bridgecrew.io/docs/ensure-iam-policies-do-not-allow-write-access-without-constraint
Check: CKV_AWS_107: "Ensure IAM policies does not allow credentials exposure"
	PASSED for resource: aws_iam_policy_document.assume_role_policy
	File: /examples/complete/data.tf:17-29
	Guide: https://docs.bridgecrew.io/docs/ensure-iam-policies-do-not-allow-credentials-exposure
Check: CKV_AWS_49: "Ensure no IAM policies documents allow "*" as a statement's actions"
	PASSED for resource: aws_iam_policy_document.monitoring
	File: /examples/complete/data.tf:31-43
	Guide: https://docs.bridgecrew.io/docs/bc_aws_iam_43
Check: CKV_AWS_108: "Ensure IAM policies does not allow data exfiltration"
	PASSED for resource: aws_iam_policy_document.monitoring
	File: /examples/complete/data.tf:31-43
	Guide: https://docs.bridgecrew.io/docs/ensure-iam-policies-do-not-allow-data-exfiltration
Check: CKV_AWS_1: "Ensure IAM policies that allow full "*-*" administrative privileges are not created"
	PASSED for resource: aws_iam_policy_document.monitoring
	File: /examples/complete/data.tf:31-43
	Guide: https://docs.bridgecrew.io/docs/iam_23
Check: CKV_AWS_110: "Ensure IAM policies does not allow privilege escalation"
	PASSED for resource: aws_iam_policy_document.monitoring
	File: /examples/complete/data.tf:31-43
	Guide: https://docs.bridgecrew.io/docs/ensure-iam-policies-does-not-allow-privilege-escalation
Check: CKV_AWS_109: "Ensure IAM policies does not allow permissions management / resource exposure without constraints"
	PASSED for resource: aws_iam_policy_document.monitoring
	File: /examples/complete/data.tf:31-43
	Guide: https://docs.bridgecrew.io/docs/ensure-iam-policies-do-not-allow-permissions-management-resource-exposure-without-constraint
Check: CKV_AWS_111: "Ensure IAM policies does not allow write access without constraints"
	PASSED for resource: aws_iam_policy_document.monitoring
	File: /examples/complete/data.tf:31-43
	Guide: https://docs.bridgecrew.io/docs/ensure-iam-policies-do-not-allow-write-access-without-constraint
Check: CKV_AWS_107: "Ensure IAM policies does not allow credentials exposure"
	PASSED for resource: aws_iam_policy_document.monitoring
	File: /examples/complete/data.tf:31-43
	Guide: https://docs.bridgecrew.io/docs/ensure-iam-policies-do-not-allow-credentials-exposure
Check: CKV_AWS_45: "Ensure no hard-coded secrets exist in lambda environment"
	PASSED for resource: aws_lambda_function.mysql
	File: /examples/complete/lambda.tf:1-26
	Guide: https://docs.bridgecrew.io/docs/bc_aws_secrets_3
Check: CKV_AWS_115: "Ensure that AWS Lambda function is configured for function-level concurrent execution limit"
	PASSED for resource: aws_lambda_function.mysql
	File: /examples/complete/lambda.tf:1-26
	Guide: https://docs.bridgecrew.io/docs/ensure-that-aws-lambda-function-is-configured-for-function-level-concurrent-execution-limit
Check: CKV_AWS_173: "Check encryption settings for Lambda environmental variable"
	PASSED for resource: aws_lambda_function.mysql
	File: /examples/complete/lambda.tf:1-26
	Guide: https://docs.bridgecrew.io/docs/bc_aws_serverless_5
Check: CKV_AWS_117: "Ensure that AWS Lambda function is configured inside a VPC"
	PASSED for resource: aws_lambda_function.mysql
	File: /examples/complete/lambda.tf:1-26
	Guide: https://docs.bridgecrew.io/docs/ensure-that-aws-lambda-function-is-configured-inside-a-vpc-1
Check: CKV_AWS_50: "X-ray tracing is enabled for Lambda"
	PASSED for resource: aws_lambda_function.mysql
	File: /examples/complete/lambda.tf:1-26
	Guide: https://docs.bridgecrew.io/docs/bc_aws_serverless_4
Check: CKV_AWS_61: "Ensure AWS IAM policy does not allow assume role permission across all services"
	PASSED for resource: aws_iam_role.lambda
	File: /examples/complete/lambda.tf:45-48
	Guide: https://docs.bridgecrew.io/docs/bc_aws_iam_45
Check: CKV_AWS_60: "Ensure IAM role allows only specific services or principals to assume it"
	PASSED for resource: aws_iam_role.lambda
	File: /examples/complete/lambda.tf:45-48
	Guide: https://docs.bridgecrew.io/docs/bc_aws_iam_44
Check: CKV_AWS_40: "Ensure IAM policies are attached only to groups or roles (Reducing access management complexity may in-turn reduce opportunity for a principal to inadvertently receive or retain excessive privileges.)"
	PASSED for resource: aws_iam_policy_attachment.lambda
	File: /examples/complete/lambda.tf:55-59
	Guide: https://docs.bridgecrew.io/docs/iam_16-iam-policy-privileges-1
Check: CKV_AWS_63: "Ensure no IAM policies documents allow "*" as a statement's actions"
	PASSED for resource: aws_iam_policy.mysql_lambda_policy
	File: /examples/complete/lambda.tf:61-65
	Guide: https://docs.bridgecrew.io/docs/iam_48
Check: CKV_AWS_62: "Ensure IAM policies that allow full "*-*" administrative privileges are not created"
	PASSED for resource: aws_iam_policy.mysql_lambda_policy
	File: /examples/complete/lambda.tf:61-65
	Guide: https://docs.bridgecrew.io/docs/iam_47
Check: CKV_AWS_40: "Ensure IAM policies are attached only to groups or roles (Reducing access management complexity may in-turn reduce opportunity for a principal to inadvertently receive or retain excessive privileges.)"
	PASSED for resource: aws_iam_policy_attachment.mysql_lambda_policy
	File: /examples/complete/lambda.tf:67-71
	Guide: https://docs.bridgecrew.io/docs/iam_16-iam-policy-privileges-1
Check: CKV_AWS_24: "Ensure no security groups allow ingress from 0.0.0.0:0 to port 22"
	PASSED for resource: aws_security_group.lambda
	File: /examples/complete/main.tf:54-83
	Guide: https://docs.bridgecrew.io/docs/networking_1-port-security
Check: CKV_AWS_25: "Ensure no security groups allow ingress from 0.0.0.0:0 to port 3389"
	PASSED for resource: aws_security_group.lambda
	File: /examples/complete/main.tf:54-83
	Guide: https://docs.bridgecrew.io/docs/networking_2
Check: CKV_AWS_260: "Ensure no security groups allow ingress from 0.0.0.0:0 to port 80"
	PASSED for resource: aws_security_group.lambda
	File: /examples/complete/main.tf:54-83
Check: CKV_AWS_23: "Ensure every security groups rule has a description"
	PASSED for resource: aws_security_group.lambda
	File: /examples/complete/main.tf:54-83
	Guide: https://docs.bridgecrew.io/docs/networking_31
Check: CKV_AWS_149: "Ensure that Secrets Manager secret is encrypted using KMS CMK"
	PASSED for resource: module.binary_string.aws_secretsmanager_secret.this
	File: /main.tf:2-12
	Calling File: /examples/binary/main.tf:6-20
	Guide: https://docs.bridgecrew.io/docs/ensure-that-secrets-manager-secret-is-encrypted-using-kms
Check: CKV_AWS_149: "Ensure that Secrets Manager secret is encrypted using KMS CMK"
	PASSED for resource: module.secret_rotation.aws_secretsmanager_secret.this
	File: /main.tf:2-12
	Calling File: /examples/complete/main.tf:2-24
	Guide: https://docs.bridgecrew.io/docs/ensure-that-secrets-manager-secret-is-encrypted-using-kms
Check: CKV_AWS_149: "Ensure that Secrets Manager secret is encrypted using KMS CMK"
	PASSED for resource: module.key_pair_string.aws_secretsmanager_secret.this
	File: /main.tf:2-12
	Calling File: /examples/key-value/main.tf:1-20
	Guide: https://docs.bridgecrew.io/docs/ensure-that-secrets-manager-secret-is-encrypted-using-kms
Check: CKV_AWS_149: "Ensure that Secrets Manager secret is encrypted using KMS CMK"
	PASSED for resource: module.minimum.aws_secretsmanager_secret.this
	File: /main.tf:2-12
	Calling File: /examples/minimum/main.tf:1-14
	Guide: https://docs.bridgecrew.io/docs/ensure-that-secrets-manager-secret-is-encrypted-using-kms
Check: CKV2_AWS_5: "Ensure that Security Groups are attached to another resource"
	PASSED for resource: aws_security_group.lambda
	File: /examples/complete/main.tf:54-83
	Guide: https://docs.bridgecrew.io/docs/ensure-that-security-groups-are-attached-to-ec2-instances-or-elastic-network-interfaces-enis
Check: CKV_AWS_116: "Ensure that AWS Lambda function is configured for a Dead Letter Queue(DLQ)"
	SKIPPED for resource: aws_lambda_function.mysql
	Suppress comment:  "Ensure that AWS Lambda function is configured for a Dead Letter Queue(DLQ)"
	File: /examples/complete/lambda.tf:1-26
	Guide: https://docs.bridgecrew.io/docs/ensure-that-aws-lambda-function-is-configured-for-a-dead-letter-queue-dlq
```
</details>

### CODE REVIEW

File: examples/complete/locals.tf [hunk 1]
```diff
@@ -29,4 +29,42 @@ locals {
...
+        Effect = "Allow",
+        Action = [
+          "secretsmanager:DescribeSecret",
+          "secretsmanager:GetSecretValue",
+          "secretsmanager:PutSecretValue",
+          "secretsmanager:UpdateSecretVersionStage",
+        ],
+        Resource = [
+          "arn:${data.aws_partition.current.partition}:secretsmanager:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:secret:*",
+        ]
```
Comment (patrickmukumbu) [inline examples/complete/locals.tf]: Use locals here. See line 25 above

File: examples/complete/mysql_db.tf [hunk 2]
```diff
@@ -1,45 +1,57 @@
...
-resource "aws_iam_role" "monitoring" {
-  name               = "${local.name}-enhanced-monitoring-role"
-  assume_role_policy = data.aws_iam_policy_document.monitoring.json
-  description        = "enhanced monitoring iam role for rds instance."
-}
+  deletion_protection                 = false
+  vpc_id                              = module.rotation_vpc.id
+  assume_role_policy                  = data.aws_iam_policy_document.monitoring.json
+  policy_arn                          = "arn:${data.aws_partition.current.partition}:iam::aws:policy/service-role/AmazonRDSEnhancedMonitoringRole"
+  major_engine_version                = "8.0"
```
Comment (patrickmukumbu) [inline examples/complete/mysql_db.tf]: Same here, use locals

Comment (patrickmukumbu) [inline examples/complete/mysql_db.tf]: Same here, use locals

Comment (patrickmukumbu) [inline examples/complete/mysql_db.tf]: Same here, use locals

**Review by patrickmukumbu — CHANGES_REQUESTED** [review]
Hey @ndegwajohn please add `supporting resouces note in changelog`. See my other comments

**Review by patrickmukumbu — CHANGES_REQUESTED** [review]
Hey @ndegwajohn please add `supporting resouces note in changelog`. See my other comments

File: examples/complete/locals.tf (hunk 1 above)
Comment (j0hnmwangi) [reply examples/complete/locals.tf]: Great point @patrickmukumbu

**Review by patrickmukumbu — APPROVED** [review]
LGTM!

**Pull request merged**